PyDSS/pyPostprocessor/PostprocessScripts/AutomatedThermalUpgrade.py

```diff
@@ -72,6 +90,10 @@ def __init__(self, project, scenario, inputs, dssInstance, dssSolver, dssObjects
         super(AutomatedThermalUpgrade, self).__init__(project, scenario, inputs, dssInstance, dssSolver, dssObjects, dssObjectsByClass, simulationSettings, Logger)
         dss = dssInstance
         self.dssSolver = dssSolver
+
+        self.orig_xfmrs = list(iter_elements(dss.Transformers, get_transformer_info))
```

Comment:
This is related to my earlier comment about not storing the outer dictionaries.  Suppose `get_transformer_info` returned a dict with the keys `name`, `kva`, `conn`, and `kv`.  This line could be changed to:

```
self.orig_xfmrs = {x["name"]: x for x in iter_elements(dss.Transformers, get_transformer_info)}
```

Then you wouldn't have to do the conversions in postprocess_voltage_upgrades.  Your code is correct as it is.  This suggestion would make it more maintainable and a bit easier to follow.

Reply:
Oh okay, yeah. I like that.